### PR TITLE
Cannot launch DirectPipeline without a Google Account selected #2925

### LINKED ISSUE #2925: Cannot launch DirectPipeline without a Google Account selected
1. Create new Dataflow project.  Do not set any account defaults.
2. Launch with _Debug As > Dataflow Pipeline_.  
   1. Select the _StarterPipeline_
   2. On the _Pipeline Arguments_ tab, select the DirectRunner and ensure the _Use default Dataflow options_ is checked.
3. Click the _Debug_ button.

screen shot 2018-03-08 at 3 27 11 pm

Selecting a Google account either in the launch configuration or the project settings is enough to get this working.

### PR DESCRIPTION
**Pull request by lak-proddev**
fixes #2925

### CODE REVIEW

**Comment by elharo** [comment]
tests are faiing

**Comment by lak-proddev** [comment]
I have changed the test case. Please review it.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/ScrolledPageContent.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ */
+
+package com.google.cloud.tools.eclipse.appengine.deploy.ui;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.ui.forms.widgets.SharedScrolledComposite;
+
+
+public class ScrolledPageContent extends SharedScrolledComposite {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/ScrolledPageContent.java]: This class probably doesn't need to be public.

File: plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java [hunk 2]
```diff
@@ -670,4 +675,9 @@ void join() throws InterruptedException {
     }
     projectInput.join();
   }
+
+  public void setAccountRequired(boolean required) {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java]: dosn't need to be public

Comment (lak-proddev) [reply plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/preferences/RunOptionsDefaultsComponent.java]: It needs to be public. this method called by  com.google.cloud.tools.eclipse.dataflow.ui.launcher.DefaultedPipelineOptionsComponent.setAccountRequired(boolean)

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/ScrolledPageContent.java [hunk 3]
```diff
@@ -21,7 +21,7 @@
 import org.eclipse.ui.forms.widgets.SharedScrolledComposite;
 
 
-public class ScrolledPageContent extends SharedScrolledComposite {
+class ScrolledPageContent extends SharedScrolledComposite {
 
   public ScrolledPageContent(Composite parent) {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/ScrolledPageContent.java]: methods/constructors don't need to be public here either

Comment (lak-proddev) [reply plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/ScrolledPageContent.java]: Actually, we don't really need a separate class here.  For the future to use in other classes I made it separately. :)

File: plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/launcher/DefaultedPipelineOptionsComponent.java [hunk 4]
```diff
@@ -236,4 +236,8 @@ public void widgetSelected(SelectionEvent e) {
   public void validate() {
     defaultOptions.validate();
   }
+
+  public void setAccountRequired(boolean required) {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/launcher/DefaultedPipelineOptionsComponent.java]: I don't think this method needs to be public. Yes, there's probably a lot of other currently public methods that don't need to be either. One of these days, I'd really like to clean up the dataflow package.

**Comment by elharo** [comment]
Merged. Thanks!

**Pull request merged**